lib/jekyll/collection.rb

```diff
@@ -204,5 +195,24 @@ def extract_metadata
         {}
       end
     end
+
+    private
+    def read_docs(full_path)
+      doc = Jekyll::Document.new(full_path, { :site => site, :collection => self })
```

Comment:
You don't like the 1-line solution there? I think it reads nicely. We could even remove the curly braces:

```ruby
doc = Document.new(full_path, :site => site, :collection => self)
```